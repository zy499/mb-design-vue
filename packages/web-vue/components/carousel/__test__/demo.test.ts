import demoTest from '../../../scripts/demo-test';

demoTest('carousel');
