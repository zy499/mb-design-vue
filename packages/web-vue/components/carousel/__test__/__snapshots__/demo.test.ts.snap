// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<carousel> demo: render [auto] correctly 1`] = `
"<div class=\\"mb-carousel mb-carousel-indicator-position-bottom\\" style=\\"width: 600px; height: 240px;\\">
  <div class=\\"mb-carousel-slide mb-carousel-horizontal\\">
    <div aria-hidden=\\"true\\" class=\\"\\" style=\\"transition-timing-function: cubic-bezier(0.34, 0.69, 0.1, 1); transition-duration: 500ms; animation-timing-function: cubic-bezier(0.34, 0.69, 0.1, 1); animation-duration: 500ms;\\"><img src=\\"https://p1-mb.byteimg.com/tos-cn-i-uwbnlip3yd/cd7a1aaea8e1c5e3d26fe2591e561798.png~tplv-uwbnlip3yd-webp.webp\\" style=\\"width: 100%;\\"></div>
    <div aria-hidden=\\"true\\" class=\\"\\" style=\\"transition-timing-function: cubic-bezier(0.34, 0.69, 0.1, 1); transition-duration: 500ms; animation-timing-function: cubic-bezier(0.34, 0.69, 0.1, 1); animation-duration: 500ms;\\"><img src=\\"https://p1-mb.byteimg.com/tos-cn-i-uwbnlip3yd/6480dbc69be1b5de95010289787d64f1.png~tplv-uwbnlip3yd-webp.webp\\" style=\\"width: 100%;\\"></div>
    <div aria-hidden=\\"true\\" class=\\"\\" style=\\"transition-timing-function: cubic-bezier(0.34, 0.69, 0.1, 1); transition-duration: 500ms; animation-timing-function: cubic-bezier(0.34, 0.69, 0.1, 1); animation-duration: 500ms;\\"><img src=\\"https://p1-mb.byteimg.com/tos-cn-i-uwbnlip3yd/0265a04fddbd77a19602a15d9d55d797.png~tplv-uwbnlip3yd-webp.webp\\" style=\\"width: 100%;\\"></div>
  </div>
  <!---->
  <!---->
</div>"
`;

exports[`<carousel> demo: render [basic] correctly 1`] = `
"<div class=\\"mb-carousel mb-carousel-indicator-position-bottom\\" style=\\"width: 600px; height: 240px;\\">
  <div class=\\"mb-carousel-slide mb-carousel-horizontal\\">
    <div aria-hidden=\\"true\\" class=\\"\\" style=\\"transition-timing-function: cubic-bezier(0.34, 0.69, 0.1, 1); transition-duration: 500ms; animation-timing-function: cubic-bezier(0.34, 0.69, 0.1, 1); animation-duration: 500ms;\\"><img src=\\"https://p1-mb.byteimg.com/tos-cn-i-uwbnlip3yd/cd7a1aaea8e1c5e3d26fe2591e561798.png~tplv-uwbnlip3yd-webp.webp\\" style=\\"width: 100%;\\"></div>
    <div aria-hidden=\\"true\\" class=\\"\\" style=\\"transition-timing-function: cubic-bezier(0.34, 0.69, 0.1, 1); transition-duration: 500ms; animation-timing-function: cubic-bezier(0.34, 0.69, 0.1, 1); animation-duration: 500ms;\\"><img src=\\"https://p1-mb.byteimg.com/tos-cn-i-uwbnlip3yd/6480dbc69be1b5de95010289787d64f1.png~tplv-uwbnlip3yd-webp.webp\\" style=\\"width: 100%;\\"></div>
    <div aria-hidden=\\"true\\" class=\\"\\" style=\\"transition-timing-function: cubic-bezier(0.34, 0.69, 0.1, 1); transition-duration: 500ms; animation-timing-function: cubic-bezier(0.34, 0.69, 0.1, 1); animation-duration: 500ms;\\"><img src=\\"https://p1-mb.byteimg.com/tos-cn-i-uwbnlip3yd/0265a04fddbd77a19602a15d9d55d797.png~tplv-uwbnlip3yd-webp.webp\\" style=\\"width: 100%;\\"></div>
  </div>
  <!---->
  <!---->
</div>"
`;

exports[`<carousel> demo: render [card] correctly 1`] = `
"<div class=\\"mb-carousel mb-carousel-indicator-position-outer\\" style=\\"width: 100%; height: 240px;\\">
  <div class=\\"mb-carousel-card mb-carousel-horizontal\\">
    <div aria-hidden=\\"true\\" class=\\"\\" style=\\"transition-timing-function: cubic-bezier(0.34, 0.69, 0.1, 1); transition-duration: 500ms; animation-timing-function: cubic-bezier(0.34, 0.69, 0.1, 1); animation-duration: 500ms; width: 60%;\\"><img src=\\"https://p1-mb.byteimg.com/tos-cn-i-uwbnlip3yd/cd7a1aaea8e1c5e3d26fe2591e561798.png~tplv-uwbnlip3yd-webp.webp\\" style=\\"width: 100%;\\"></div>
    <div aria-hidden=\\"true\\" class=\\"\\" style=\\"transition-timing-function: cubic-bezier(0.34, 0.69, 0.1, 1); transition-duration: 500ms; animation-timing-function: cubic-bezier(0.34, 0.69, 0.1, 1); animation-duration: 500ms; width: 60%;\\"><img src=\\"https://p1-mb.byteimg.com/tos-cn-i-uwbnlip3yd/6480dbc69be1b5de95010289787d64f1.png~tplv-uwbnlip3yd-webp.webp\\" style=\\"width: 100%;\\"></div>
    <div aria-hidden=\\"true\\" class=\\"\\" style=\\"transition-timing-function: cubic-bezier(0.34, 0.69, 0.1, 1); transition-duration: 500ms; animation-timing-function: cubic-bezier(0.34, 0.69, 0.1, 1); animation-duration: 500ms; width: 60%;\\"><img src=\\"https://p1-mb.byteimg.com/tos-cn-i-uwbnlip3yd/0265a04fddbd77a19602a15d9d55d797.png~tplv-uwbnlip3yd-webp.webp\\" style=\\"width: 100%;\\"></div>
    <div aria-hidden=\\"true\\" class=\\"\\" style=\\"transition-timing-function: cubic-bezier(0.34, 0.69, 0.1, 1); transition-duration: 500ms; animation-timing-function: cubic-bezier(0.34, 0.69, 0.1, 1); animation-duration: 500ms; width: 60%;\\"><img src=\\"https://p1-mb.byteimg.com/tos-cn-i-uwbnlip3yd/24e0dd27418d2291b65db1b21aa62254.png~tplv-uwbnlip3yd-webp.webp\\" style=\\"width: 100%;\\"></div>
  </div>
  <!---->
  <!---->
</div>"
`;

exports[`<carousel> demo: render [direction] correctly 1`] = `
"<div class=\\"mb-carousel mb-carousel-indicator-position-right\\" style=\\"width: 600px; height: 240px;\\">
  <div class=\\"mb-carousel-slide mb-carousel-vertical\\">
    <div aria-hidden=\\"true\\" class=\\"\\" style=\\"transition-timing-function: cubic-bezier(0.34, 0.69, 0.1, 1); transition-duration: 500ms; animation-timing-function: cubic-bezier(0.34, 0.69, 0.1, 1); animation-duration: 500ms;\\"><img src=\\"https://p1-mb.byteimg.com/tos-cn-i-uwbnlip3yd/cd7a1aaea8e1c5e3d26fe2591e561798.png~tplv-uwbnlip3yd-webp.webp\\" style=\\"width: 100%;\\"></div>
    <div aria-hidden=\\"true\\" class=\\"\\" style=\\"transition-timing-function: cubic-bezier(0.34, 0.69, 0.1, 1); transition-duration: 500ms; animation-timing-function: cubic-bezier(0.34, 0.69, 0.1, 1); animation-duration: 500ms;\\"><img src=\\"https://p1-mb.byteimg.com/tos-cn-i-uwbnlip3yd/6480dbc69be1b5de95010289787d64f1.png~tplv-uwbnlip3yd-webp.webp\\" style=\\"width: 100%;\\"></div>
    <div aria-hidden=\\"true\\" class=\\"\\" style=\\"transition-timing-function: cubic-bezier(0.34, 0.69, 0.1, 1); transition-duration: 500ms; animation-timing-function: cubic-bezier(0.34, 0.69, 0.1, 1); animation-duration: 500ms;\\"><img src=\\"https://p1-mb.byteimg.com/tos-cn-i-uwbnlip3yd/0265a04fddbd77a19602a15d9d55d797.png~tplv-uwbnlip3yd-webp.webp\\" style=\\"width: 100%;\\"></div>
  </div>
  <!---->
  <!---->
</div>"
`;

exports[`<carousel> demo: render [fade] correctly 1`] = `
"<div class=\\"mb-carousel mb-carousel-indicator-position-bottom\\" style=\\"width: 600px; height: 240px;\\">
  <div class=\\"mb-carousel-fade mb-carousel-horizontal\\">
    <div aria-hidden=\\"true\\" class=\\"\\" style=\\"transition-timing-function: cubic-bezier(0.34, 0.69, 0.1, 1); transition-duration: 500ms; animation-timing-function: cubic-bezier(0.34, 0.69, 0.1, 1); animation-duration: 500ms;\\"><img src=\\"https://p1-mb.byteimg.com/tos-cn-i-uwbnlip3yd/cd7a1aaea8e1c5e3d26fe2591e561798.png~tplv-uwbnlip3yd-webp.webp\\" style=\\"width: 100%;\\"></div>
    <div aria-hidden=\\"true\\" class=\\"\\" style=\\"transition-timing-function: cubic-bezier(0.34, 0.69, 0.1, 1); transition-duration: 500ms; animation-timing-function: cubic-bezier(0.34, 0.69, 0.1, 1); animation-duration: 500ms;\\"><img src=\\"https://p1-mb.byteimg.com/tos-cn-i-uwbnlip3yd/6480dbc69be1b5de95010289787d64f1.png~tplv-uwbnlip3yd-webp.webp\\" style=\\"width: 100%;\\"></div>
    <div aria-hidden=\\"true\\" class=\\"\\" style=\\"transition-timing-function: cubic-bezier(0.34, 0.69, 0.1, 1); transition-duration: 500ms; animation-timing-function: cubic-bezier(0.34, 0.69, 0.1, 1); animation-duration: 500ms;\\"><img src=\\"https://p1-mb.byteimg.com/tos-cn-i-uwbnlip3yd/0265a04fddbd77a19602a15d9d55d797.png~tplv-uwbnlip3yd-webp.webp\\" style=\\"width: 100%;\\"></div>
  </div>
  <!---->
  <!---->
</div>"
`;

exports[`<carousel> demo: render [indicator] correctly 1`] = `
"<div class=\\"mb-space mb-space-vertical\\">
  <div class=\\"mb-space-item\\" style=\\"margin-bottom: 24px;\\"><span class=\\"mb-radio-group-button mb-radio-group-size-undefined mb-radio-group-direction-horizontal\\"><label class=\\"mb-radio-button mb-radio-checked\\"><input type=\\"radio\\" class=\\"mb-radio-target\\" value=\\"dot\\"><span class=\\"mb-radio-button-content\\">dot</span></label><label class=\\"mb-radio-button\\"><input type=\\"radio\\" class=\\"mb-radio-target\\" value=\\"line\\"><span class=\\"mb-radio-button-content\\">line</span></label><label class=\\"mb-radio-button\\"><input type=\\"radio\\" class=\\"mb-radio-target\\" value=\\"slider\\"><span class=\\"mb-radio-button-content\\">slider</span></label></span></div>
  <div class=\\"mb-space-item\\" style=\\"margin-bottom: 24px;\\"><span class=\\"mb-radio-group-button mb-radio-group-size-undefined mb-radio-group-direction-horizontal\\" style=\\"margin-bottom: 20px;\\"><label class=\\"mb-radio-button\\"><input type=\\"radio\\" class=\\"mb-radio-target\\" value=\\"left\\"><span class=\\"mb-radio-button-content\\">left</span></label><label class=\\"mb-radio-button\\"><input type=\\"radio\\" class=\\"mb-radio-target\\" value=\\"right\\"><span class=\\"mb-radio-button-content\\">right</span></label><label class=\\"mb-radio-button\\"><input type=\\"radio\\" class=\\"mb-radio-target\\" value=\\"top\\"><span class=\\"mb-radio-button-content\\">top</span></label><label class=\\"mb-radio-button mb-radio-checked\\"><input type=\\"radio\\" class=\\"mb-radio-target\\" value=\\"bottom\\"><span class=\\"mb-radio-button-content\\">bottom</span></label><label class=\\"mb-radio-button\\"><input type=\\"radio\\" class=\\"mb-radio-target\\" value=\\"outer\\"><span class=\\"mb-radio-button-content\\">outer</span></label></span></div>
  <div class=\\"mb-space-item\\">
    <div class=\\"mb-carousel mb-carousel-indicator-position-bottom\\" style=\\"width: 600px; height: 240px;\\">
      <div class=\\"mb-carousel-slide mb-carousel-horizontal\\">
        <div aria-hidden=\\"true\\" class=\\"\\" style=\\"transition-timing-function: cubic-bezier(0.34, 0.69, 0.1, 1); transition-duration: 500ms; animation-timing-function: cubic-bezier(0.34, 0.69, 0.1, 1); animation-duration: 500ms;\\"><img src=\\"https://p1-mb.byteimg.com/tos-cn-i-uwbnlip3yd/cd7a1aaea8e1c5e3d26fe2591e561798.png~tplv-uwbnlip3yd-webp.webp\\" style=\\"width: 100%;\\"></div>
        <div aria-hidden=\\"true\\" class=\\"\\" style=\\"transition-timing-function: cubic-bezier(0.34, 0.69, 0.1, 1); transition-duration: 500ms; animation-timing-function: cubic-bezier(0.34, 0.69, 0.1, 1); animation-duration: 500ms;\\"><img src=\\"https://p1-mb.byteimg.com/tos-cn-i-uwbnlip3yd/6480dbc69be1b5de95010289787d64f1.png~tplv-uwbnlip3yd-webp.webp\\" style=\\"width: 100%;\\"></div>
        <div aria-hidden=\\"true\\" class=\\"\\" style=\\"transition-timing-function: cubic-bezier(0.34, 0.69, 0.1, 1); transition-duration: 500ms; animation-timing-function: cubic-bezier(0.34, 0.69, 0.1, 1); animation-duration: 500ms;\\"><img src=\\"https://p1-mb.byteimg.com/tos-cn-i-uwbnlip3yd/0265a04fddbd77a19602a15d9d55d797.png~tplv-uwbnlip3yd-webp.webp\\" style=\\"width: 100%;\\"></div>
      </div>
      <!---->
      <!---->
    </div>
  </div>
</div>"
`;
